diff --git a/video/out/hwdec/hwdec_aimagereader.c b/video/out/hwdec/hwdec_aimagereader.c
index a18ef8f1f..99d3a224c 100644
--- a/video/out/hwdec/hwdec_aimagereader.c
+++ b/video/out/hwdec/hwdec_aimagereader.c
@@ -144,7 +144,7 @@ static int init(struct ra_hwdec *hw)
     // dummy dimensions, AImageReader only transports hardware buffers
     media_status_t ret = p->AImageReader_newWithUsage(16, 16,
         AIMAGE_FORMAT_PRIVATE, AHARDWAREBUFFER_USAGE_GPU_SAMPLED_IMAGE,
-        5, &p->reader);
+        2, &p->reader);
     if (ret != AMEDIA_OK) {
         MP_ERR(hw, "newWithUsage failed: %d\n", ret);
         return -1;
