diff --git a/video/out/hwdec/hwdec_aimagereader.c b/video/out/hwdec/hwdec_aimagereader.c
--- a/video/out/hwdec/hwdec_aimagereader.c
+++ b/video/out/hwdec/hwdec_aimagereader.c
@@ -157,7 +157,7 @@ static int init(struct ra_hwdec *hw)
     // (incl. ones currently queued) at the same time.
     // The documentation recommends a margin of 2 for acquireLatestImage to
     // work correctly.
-    const int max_images = 3;
+    const int max_images = 2;
     // dummy dimensions, AImageReader only transports hardware buffers
     media_status_t ret = p->AImageReader_newWithUsage(16, 16,
         AIMAGE_FORMAT_PRIVATE, AHARDWAREBUFFER_USAGE_GPU_SAMPLED_IMAGE,
